funcx_endpoint/funcx_endpoint/executors/high_throughput/funcx_manager.py

```diff
@@ -417,17 +486,18 @@ def poll_funcx_task_socket(self, test=False):
                 self.worker_map.register_worker(w_id, reg_info['worker_type'])
 
             elif m_type == b'TASK_RET':
+                # We lock because the following steps are also shared by task_cancel
+                self.task_cancel_lock.acquire()
                 logger.debug("Result received from worker: {}".format(w_id))
-                logger.debug("[TASK_PULL_THREAD] Got result: {}".format(message))
-                self.pending_result_queue.put(message)
-                self.worker_map.put_worker(w_id)
-                self.task_done_counter += 1
                 task_id = pickle.loads(message)['task_id']
-                task_type = self.task_type_mapping.pop(task_id)
-                self.task_status_deltas.pop(task_id, None)
-                logger.debug("Task type: {}".format(task_type))
-                self.outstanding_task_count[task_type] -= 1
-                logger.debug("Got result: Outstanding task counts: {}".format(self.outstanding_task_count))
+                try:
+                    self.remove_task(task_id)
+                except KeyError:
+                    logger.exception(f"Task:{task_id} missing in task structure")
```

Comment:
what are the circumstances that this would be missing in the task structure? It shouldn't have been removed by cancellation code because that is locked, right?